kivymd/uix/textfield.py

```diff
@@ -1183,7 +1183,7 @@ def set_text(self, instance, text):
             self._anim_current_line_color(self.line_color_focus)
             if self.helper_text_mode == "on_error":
                 self._anim_current_error_color((0, 0, 0, 0))
-                self.on_focus(self, self.focus)
+            self.on_focus(self, self.focus)
```

Comment:
Are you sure this change is not disruptive to the class as a whole? You have not provided any description of this change ... If you believe this change is necessary, please provide example code that need this fix.